File: patch_prices.py
import httpx, json
from pathlib import Path

data_path = Path(r'C:\Users\kanaw\.openclaw\workspace\polymarket-dashboard\data.json')
data = json.loads(data_path.read_text())

def get_price(token_id):
    if not token_id:
        return None
    try:
        r = httpx.get(f'https://clob.polymarket.com/last-trade-price?token_id={token_id}', timeout=8)
        p = float(r.json().get('price', 0))
        if p > 0:
            return p
    except:
        pass
    try:
        r = httpx.get(f'https://clob.polymarket.com/midpoint?token_id={token_id}', timeout=8)
        p = float(r.json().get('mid', 0))
        if p > 0:
            return p
    except:
        pass
    return None

all_positions = data.get('open_orders', []) + data.get('filled_positions', [])
for pos in all_positions:
    tid = pos.get('token_id')
    cur = get_price(tid)
    pos['current_price'] = cur
    if cur is not None and pos.get('filled', 0) > 0:
        entry = float(pos['price'])
        filled = float(pos['filled'])
        pos['unrealized_pnl'] = round((cur - entry) * filled, 4)
    else:
        pos['unrealized_pnl'] = None
    print(f"{pos['label']:35s}  entry={pos['price']:.3f}  now={cur}  upnl={pos.get('unrealized_pnl')}")

for bet in data.get('ufc_bets', []):
    tid = bet.get('token_id')
    cur = get_price(tid)
    bet['current_price'] = cur
    if cur is not None and bet.get('shares', 0) > 0:
        entry = float(bet['price'])
        shares = float(bet['shares'])
        bet['unrealized_pnl'] = round((cur - entry) * shares, 4)
    else:
        bet['unrealized_pnl'] = None
    print(f"{bet['fighter']:35s}  entry={bet['price']:.3f}  now={cur}  upnl={bet.get('unrealized_pnl')}")

data_path.write_text(json.dumps(data, indent=2))
print('\ndata.json updated!')

File: patch_prices.mjs
import { readFileSync, writeFileSync } from 'fs';

const dataPath = new URL('./data.json', import.meta.url).pathname.replace(/^\//, '');

const data = JSON.parse(readFileSync(dataPath, 'utf8'));

async function getPrice(tokenId) {
  if (!tokenId) return null;
  try {
    const r = await fetch(`https://clob.polymarket.com/last-trade-price?token_id=${tokenId}`);
    const d = await r.json();
    const p = parseFloat(d.price || 0);
    if (p > 0) return p;
  } catch {}
  try {
    const r = await fetch(`https://clob.polymarket.com/midpoint?token_id=${tokenId}`);
    const d = await r.json();
    const p = parseFloat(d.mid || 0);
    if (p > 0) return p;
  } catch {}
  return null;
}

const allPositions = [...(data.open_orders || []), ...(data.filled_positions || [])];
for (const pos of allPositions) {
  const cur = await getPrice(pos.token_id);
  pos.current_price = cur;
  if (cur !== null && (pos.filled || 0) > 0) {
    pos.unrealized_pnl = Math.round((cur - pos.price) * pos.filled * 10000) / 10000;
  } else {
    pos.unrealized_pnl = null;
  }
  console.log(`${(pos.label || '').padEnd(35)}  entry=${pos.price?.toFixed(3)}  now=${cur}  upnl=${pos.unrealized_pnl}`);
}

for (const bet of (data.ufc_bets || [])) {
  const cur = await getPrice(bet.token_id);
  bet.current_price = cur;
  if (cur !== null && (bet.shares || 0) > 0) {
    bet.unrealized_pnl = Math.round((cur - bet.price) * bet.shares * 10000) / 10000;
  } else {
    bet.unrealized_pnl = null;
  }
  console.log(`${(bet.fighter || '').padEnd(35)}  entry=${bet.price?.toFixed(3)}  now=${cur}  upnl=${bet.unrealized_pnl}`);
}

writeFileSync(dataPath, JSON.stringify(data, null, 2));
console.log('\ndata.json updated!');
